spec/index.md

```diff
@@ -3605,6 +3630,132 @@ S with
 
 ```
 
+
+#### IC Management Canister: Upload Chunk
+
+A controller of a canister, or the canister itself can upload chunks to the chunk store of that canister.
+
+Conditions
+
+```html
+
+S.messages = Older_messages · CallMessage M · Younger_messages
+(M.queue = Unordered) or (∀ msg ∈ Older_messages. msg.queue ≠ M.queue)
+M.method_name = 'upload_chunk'
+M.arg = candid(A)
+chunk_store_size = |{x | chunk_store[A.canister_id][x] not null}|
+chunk_store_size < CHUNK_STORE_SIZE and (M.caller ∈ S.controllers[A.canister_id] ∪ {A.canister_id})
+hash = SHA-256(A.chunk)
+M.caller ∈ S.controllers[A.canister_id] ∪ {A.canister_id}
+
+
+```
+
+State after
+
+```html
+
+S with
+    chunk_store[A.canister_id](hash') = chunk_store[A.canister_id] if hash'≠ hash
+    chunk_store[A.canister_id](hash) = A.chunk
+    messages = Older_messages · Younger_messages ·
+      ResponseMessage {
+        origin = M.origin
+        response = candid(hash)
+      }
+
+```
+
+#### IC Management Canister: Clear chunk store
+
+The controller of a canister, or the canister itself can clear the chunk store of that canister. 
+
+```html
+
+S.messages = Older_messages · CallMessage M · Younger_messages
+(M.queue = Unordered) or (∀ msg ∈ Older_messages. msg.queue ≠ M.queue)
+M.method_name = 'clear_store'
+M.arg = candid(A)
+M.caller ∈ S.controllers[A.canister_id] ∪ {A.canister_id}
+```
+
+State after
+
+```html
+
+S with
+    chunk_store[A.canister_id] = ()
+    messages = Older_messages · Younger_messages ·
+      ResponseMessage {
+        origin = M.origin
+        response = candid()
+      }
+
+```
+
+
+#### IC Management Canister: Delete chunks
+
+The controller of a canister, or the canister itself can delete chunks from the canister chunk store. 
+
+```html
+
+S.messages = Older_messages · CallMessage M · Younger_messages
+(M.queue = Unordered) or (∀ msg ∈ Older_messages. msg.queue ≠ M.queue)
+M.method_name = 'delete_chunks'
+M.arg = candid(A)
+chunk_store'[hash] = chunk_store[A.canister_id][hash] if hash ∉ A.hash_list
+chunk_store'[hash] = null if hash ∈ A.hash_list
```

Comment:
I'd drop these lines and undefine `hash` in the "State after" section analogously to the canister deletion transition (for instance).